src/routes/safe/components/Transactions/TxList/MethodDetails.tsx

```diff
@@ -1,44 +1,32 @@
-import { Text } from '@gnosis.pm/safe-react-components'
-
 import styled from 'styled-components'
 
-import { isArrayParameter } from 'src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/utils'
-import Value from 'src/routes/safe/components/Transactions/TxList/MethodValue'
 import { DataDecoded } from '@gnosis.pm/safe-react-gateway-sdk'
-
-const TxDetailsMethodParam = styled.div<{ isArrayParameter: boolean }>`
-  padding-left: 24px;
-  display: ${({ isArrayParameter }) => (isArrayParameter ? 'block' : 'flex')};
-  align-items: center;
-
-  p:first-of-type {
-    margin-right: ${({ isArrayParameter }) => (isArrayParameter ? '0' : '4px')};
-  }
-`
+import { StyledDetailsTitle } from 'src/routes/safe/components/Transactions/TxList/styled'
+import { TxDataRow } from 'src/routes/safe/components/Transactions/TxList/TxDataRow'
+import { camelCaseToSpaces } from 'src/routes/safe/components/Transactions/TxList/utils'
 
 const TxInfo = styled.div`
   padding: 8px 0;
 `
 
-const StyledMethodName = styled(Text)`
-  white-space: nowrap;
-`
-
 export const MethodDetails = ({ data }: { data: DataDecoded }): React.ReactElement => {
+  const methodName = camelCaseToSpaces(data.method)
```

Comment:
This doesn't cover certain edge cases like `swapExactETHForTokens` -> `swap Exact E T H For Tokens`
We could try only matching uppercase characters that aren't directly followed by another uppercase character.

Reply:
solved it 👌